invenio_records_rest/__init__.py

```diff
@@ -22,7 +22,634 @@
+
+Elasticsearch
+~~~~~~~~~~~~~
+
+Records are represented as JSON documents internally in Elasticsearch and
+grouped under an index specified in the configuration, an index represents a
+collection of documents with similar characteristics. Because the shape of a
+record can change with time, there is the possibility to group indeces with
+the same meaning but with different versions under an alias. Besides, aliases
+can help for grouping and easy filtering purposed. The structure, internal
+fields of the documents, along with indexing preferences can be defined by
+mappings, applied to specific indices. A mapping is a JSON file, which is
+loaded during initialization by Invenio-Search from a path defined by an
+entrypoint. The layout of a mapping can highly affect the search and indexing
+performance.
+
+Initialization
+--------------
+
+First create a Flask application:
+
+>>> from flask import Flask
+>>> app = Flask('myapp')
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+>>> app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+
+Initialize the required dependencies:
+
+>>> from invenio_db import InvenioDB
+>>> from invenio_rest import InvenioREST
+>>> from invenio_pidstore import InvenioPIDStore
+>>> from invenio_records import InvenioRecords
+>>> from invenio_records_rest import InvenioRecordsREST
+>>> from invenio_search import InvenioSearch
+>>> from invenio_indexer import InvenioIndexer
+>>> from invenio_records_rest.utils import PIDConverter
+>>> ext_db = InvenioDB(app)
+>>> ext_rest = InvenioREST(app)
+>>> ext_pidstore = InvenioPIDStore(app)
+>>> ext_records = InvenioRecords(app)
+>>> ext_search = InvenioSearch(app)
+>>> ext_indexer = InvenioIndexer(app)
+>>> app.url_map.converters['pid'] = PIDConverter
+
+In order for the following examples to work, you need to work within a Flask
+application context so let’s push one:
+
+>>> ctx = app.app_context()
+>>> ctx.push()
+
+Also, for the examples to work we need to create the database and tables (note
+that in this example we use an in-memory SQLite database):
+
+>>> from invenio_db import db
+>>> db.create_all()
+
+Configuration
+-------------
+
+`Namespaces` are an important concept in Invenio-Records-REST. You can think of
+them as a way of grouping related metadata.
+Imagine a `namespace` called ``records`` and another ``authors``. You could
+use ``records`` to provide bibliographic metadata and ``authors`` for author
+metadata. A namespaces is mapped one to one to Elasticsearch indeces or
+aliases.
+
+In Invenio-Records-REST these namespaces are mapped to endpoints (i.e.,
+``/records/`` or ``/authors/``). Inside these namespaces you can make
+unique reference to specific records thanks to the persistent identifiers
+or ``pid``s. Going back to our general example, a persistent identifier
+for our namespace ``authors`` could be an `ORCID <https://orcid.org/>`_ and a
+specific author would be identified by an endpoint like
+``/authors/<AUTHOR_ORCID>``.
```

Comment:
Maybe use a real ORCID here like the example [0000-0002-1825-0097](https://orcid.org/0000-0002-1825-0097)